core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -2317,10 +2318,10 @@ public void finalizeUpdateOfExchangeSinkInstanceHandle(TaskId taskId, ExchangeSi
             partition.updateExchangeSinkInstanceHandle(taskId, updatedExchangeSinkInstanceHandle);
         }
 
-        public void taskFinished(TaskId taskId, TaskStatus taskStatus)
```

Comment:
post mortem: as it gets more complicated for unfamiliar reader, it would be good to add some javadoc what does it return as @dekimir suggested

Reply:
Ok - as you both want that I will add a comment.

Reply:
https://github.com/trinodb/trino/pull/22500